booking-app/src/main/java/org/motechproject/bookingapp/domain/VisitBookingDetails.java

```diff
@@ -43,6 +45,36 @@ public VisitBookingDetails(Visit visit, LocalDate bookingActualDate) {
         this.bookingActualDate = bookingActualDate;
     }
 
+    public VisitBookingDetailsDto toDto() {
+
+        Subject participant = getVisit().getSubject();
+
+        VisitBookingDetailsDto dto = new VisitBookingDetailsDto();
+        dto.setBookingId(getId());
+        dto.setParticipantId(participant.getSubjectId());
+        dto.setParticipantName(participant.getName() + " " + participant.getHouseholdName());
+        dto.setFemaleChildBearingAge(isFemaleChildBearingAge(participant));
+
+        for (Visit participantVisit : participant.getVisits()) {
```

Comment:
if you using this only with Screening details you don't need to iterate through all visits, just use getVisit() instead